src/Lib/V1/Contexts.ts

```diff
@@ -379,3 +396,301 @@ export const isPlutusDataTxInfo: IsPlutusData<TxInfo> = {
     throw new IsPlutusDataError("Unexpected data");
   },
 };
+
+/**
+ * {@link ScriptPurpose} Purpose of the script that is currently running
+ *
+ * @see {@link https://github.com/IntersectMBO/plutus/blob/1.16.0.0/plutus-ledger-api/src/PlutusLedgerApi/V1/Contexts.hs#L79-L84 }
+ */
+export type ScriptPurpose =
+  | { name: "Minting"; fields: CurrencySymbol }
+  | { name: "Spending"; fields: TxOutRef }
+  | { name: "Rewarding"; fields: StakingCredential }
+  | { name: "Certifying"; fields: DCert };
+
+/**
+ * {@link Eq} instance for {@link ScriptPurpose}
+ */
+export const eqScriptPurpose: Eq<ScriptPurpose> = {
+  eq: (l, r) => {
+    if (l.name === "Minting" && r.name === "Minting") {
+      return LbValue.eqCurrencySymbol.eq(l.fields, r.fields);
+    } else if (l.name === "Spending" && r.name === "Spending") {
+      return LbTx.eqTxOutRef.eq(l.fields, r.fields);
+    } else if (l.name === "Rewarding" && r.name === "Rewarding") {
+      return LbCredential.eqStakingCredential.eq(l.fields, r.fields);
+    } else if (l.name === "Certifying" && r.name === "Certifying") {
+      return LbDCert.eqDCert.eq(l.fields, r.fields);
+    } else {
+      return false;
+    }
+  },
+  neq: (l, r) => {
+    if (l.name === "Minting" && r.name === "Minting") {
+      return LbValue.eqCurrencySymbol.neq(l.fields, r.fields);
+    } else if (l.name === "Spending" && r.name === "Spending") {
+      return LbTx.eqTxOutRef.neq(l.fields, r.fields);
+    } else if (l.name === "Rewarding" && r.name === "Rewarding") {
+      return LbCredential.eqStakingCredential.neq(l.fields, r.fields);
+    } else if (l.name === "Certifying" && r.name === "Certifying") {
+      return LbDCert.eqDCert.neq(l.fields, r.fields);
+    } else {
+      return true;
+    }
+  },
+};
+
+/**
+ * {@link Json} instance for {@link ScriptPurpose}
+ */
+export const jsonScriptPurpose: Json<ScriptPurpose> = {
+  toJson: (scriptPurpose) => {
+    switch (scriptPurpose.name) {
+      case "Minting":
+        return Prelude.jsonConstructor(scriptPurpose.name, [
+          LbValue.jsonCurrencySymbol.toJson(scriptPurpose.fields),
+        ]);
+      case "Spending":
+        return Prelude.jsonConstructor(scriptPurpose.name, [
+          LbTx.jsonTxOutRef.toJson(scriptPurpose.fields),
+        ]);
+      case "Rewarding":
+        return Prelude.jsonConstructor(scriptPurpose.name, [
+          LbCredential.jsonStakingCredential.toJson(scriptPurpose.fields),
+        ]);
+      case "Certifying":
+        return Prelude.jsonConstructor(scriptPurpose.name, [
+          LbDCert.jsonDCert.toJson(scriptPurpose.fields),
+        ]);
+    }
+  },
+  fromJson: (value) => {
+    return Prelude.caseJsonConstructor<ScriptPurpose>(
+      "Plutus,V1.ScriptPurpose",
+      {
+        Minting: (ctorFields) => {
+          if (ctorFields.length !== 1) {
+            throw new JsonError("Expected one field");
+          }
+          return {
+            name: "Minting",
```

Comment:
Please put `fields` before `name` i.e., something like

```
{
            fields: LbValue.jsonCurrencySymbol.fromJson(ctorFields[0]!),
            name: "Minting",
}
```

Iirc, LambdaBuffers integration test suite for equality is a bit too strict and doesn't realize that these are the same JSON objects (i.e., order of keys doesn't change equality of JSON objects) as the golden tests generated by Haskell order the keys lexicographically whereas JS orders them on insertion order.

Or alternatively, you could modify the integration test suite to loosen the notion of equality.

If you choose the former, please update all instances s.t. the keys are ordered lexicographically s.t. it matches Haskell's output when converted to a string.

Reply:
I chose the easier option, which is the latter one. The instance of v1 TxInfo is fixed as well.